File: board-room/src/main/java/net/stzups/board/room/server/websocket/protocol/server/messages/ServerMessageDeleteDocument.java
package net.stzups.board.room.server.websocket.protocol.server.messages;

import io.netty.buffer.ByteBuf;
import net.stzups.board.data.objects.Document;
import net.stzups.board.room.server.websocket.protocol.server.ServerMessage;
import net.stzups.board.room.server.websocket.protocol.server.ServerMessageType;

public class ServerMessageDeleteDocument extends ServerMessage {
    private final long id;

    public ServerMessageDeleteDocument(Document document) {
        super(ServerMessageType.DELETE_DOCUMENT);
        id = document.getId();
    }

    @Override
    public void serialize(ByteBuf byteBuf) {
        super.serialize(byteBuf);
        byteBuf.writeLong(id);
    }
}

File: scribbleshare-app/scribbleshare-frontend/src/scripts/common/protocol/server/messages/ServerMessageDeleteDocument.ts
import ServerMessage from "../ServerMessage.js";
import ServerMessageType from "../ServerMessageType.js";
import ByteBuffer from "../../ByteBuffer.js";

export default class ServerMessageDeleteDocument extends ServerMessage {
    id: bigint;

    constructor(byteBuffer: ByteBuffer) {
        super();
        this.id = byteBuffer.readBigInt64();
    }

    getType(): ServerMessageType {
        return ServerMessageType.DELETE_DOCUMENT;
    }
}